compiler/rustc_attr_data_structures/src/attributes.rs

```diff
@@ -37,11 +37,18 @@ pub enum InstructionSetAttr {
 
 #[derive(Clone, Encodable, Decodable, Debug, HashStable_Generic)]
 pub enum OptimizeAttr {
-    None,
+    /// `#[optimize(none)]` - NOT the absence of an `#[optimize(..)]` attribute
+    DoNotOptimize,
```

Comment:
Why have you deleted the old `None`? It feels like wrapping this in an `Option` does not really make anything better.

I'd leave the `None` variant, but renamed to `Default`.

Reply:
The variant was initially called `None` and renamed to this after a review comment. I felt that it made morse sense to have the absence of an `#[optimize(..)]` stored as the absence of a `OptimizeAttr`

Reply:
I do see the point, but imo `Option` just adds clutter to the handling code. Most matches I've seen match on all 4 options anyway, so the `Option` does help simplify the code.